ui/desktop/src/components/SessionHistoryView.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useMemo } from 'react';
+import { Clock, MessageSquare, ArrowLeft, AlertCircle } from 'lucide-react';
+import { type SessionDetails } from '../api/sessions';
+import { Card } from './ui/card';
+import { Button } from './ui/button';
+import BackButton from './ui/BackButton';
+import { ScrollArea } from './ui/scroll-area';
+import MarkdownContent from './MarkdownContent';
+import ToolCallWithResponse from './ToolCallWithResponse';
+import { ToolRequestMessageContent, ToolResponseMessageContent } from '../types/message';
+
+interface SessionHistoryViewProps {
+  session: SessionDetails;
+  isLoading: boolean;
+  error: string | null;
+  onBack: () => void;
+  onResume: () => void;
+  onRetry: () => void;
+}
+
+const SessionHistoryView: React.FC<SessionHistoryViewProps> = ({
+  session,
+  isLoading,
+  error,
+  onBack,
+  onResume,
+  onRetry,
+}) => {
+  // Move the tool response mapping logic outside of the render loop
+  const getToolResponsesMap = (messageIndex: number, toolRequests: ToolRequestMessageContent[]) => {
```

Comment:
Could move the definition of this outside the react component entirely. I think you'd just need to pass in `session`